underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystem.java

```diff
@@ -444,34 +444,38 @@ public long getSpace(String path, SpaceType type) throws IOException {
     // as Alluxio can load/store data out of entire HDFS cluster
     FileSystem hdfs = getFs();
     long space = -1;
-    if (hdfs instanceof DistributedFileSystem) {
-      // Note that, getDiskStatus() is an API from Hadoop 1, deprecated by getStatus() from
-      // Hadoop 2 and removed in Hadoop 3
-      switch (type) {
-        case SPACE_TOTAL:
-          //#ifdef HADOOP1
+    // Note that, getDiskStatus() is an API from Hadoop 1, deprecated by getStatus() from
+    // Hadoop 2 and removed in Hadoop 3
+    switch (type) {
+      case SPACE_TOTAL:
+        //#ifdef HADOOP1
+        if (hdfs instanceof DistributedFileSystem) {
           space = ((DistributedFileSystem) hdfs).getDiskStatus().getCapacity();
-          //#else
-          space = hdfs.getStatus().getCapacity();
-          //#endif
-          break;
-        case SPACE_USED:
-          //#ifdef HADOOP1
+        }
+        //#else
+        space = hdfs.getStatus().getCapacity();
+        //#endif
+        break;
+      case SPACE_USED:
+        //#ifdef HADOOP1
+        if (hdfs instanceof DistributedFileSystem) {
           space = ((DistributedFileSystem) hdfs).getDiskStatus().getDfsUsed();
-          //#else
-          space = hdfs.getStatus().getUsed();
-          //#endif
-          break;
-        case SPACE_FREE:
-          //#ifdef HADOOP1
+        }
+        //#else
+        space = hdfs.getStatus().getUsed();
+        //#endif
+        break;
+      case SPACE_FREE:
+        //#ifdef HADOOP1
+        if (hdfs instanceof DistributedFileSystem) {
           space = ((DistributedFileSystem) hdfs).getDiskStatus().getRemaining();
-          //#else
-          space = hdfs.getStatus().getRemaining();
-          //#endif
-          break;
-        default:
-          throw new IOException("Unknown space type: " + type);
-      }
+        }
+        //#else
+        space = hdfs.getStatus().getRemaining();
```

Comment:
what is changed here?

Reply:
Previously, only DistributedFileSystem could be processed. But OzoneFileSystem is not an instance of DistributedFileSystem, so it will not get value, and get default value -1. And this module is not relay on Ozone jar, can not write `if instance of OzoneFileSystem`.   And I let all filesystems can  obtain the value except for hadoop1 which remain  original logic.

Reply:
Moved the comment here from flaming-archer

For hadoop 2 and 3, if HDFS is not DistributedFileSystem, does it have those APIs (e.g. hdfs.getStatus().getCapacity();)?----- Hadoop2 has these interfaces, the original use this interface and we use the version of hadoop2. As for hadoop3 version, I checked that there are also these interfaces:
https://github.com/apache/hadoop/blob/release-3.3.5-RC0/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileSystem.java
https://github.com/apache/hadoop/blob/release-3.3.5-RC0/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FsStatus.java

Ozone has, but are there any other filesystem that is not Ozone and is not DistributedFileSystem does not have those APIs?-----Because this interface is a hadoop interface, as long as they are compatible with hadoop, they must have this interface.

Reply:
Yes. Hadoop 2 and hadoop 3 both support the following APIS. And as long as the target hadoop file system is hadoop compatible, it should follow the convention to support those APIs